magento/magento2
# Initialize toolbar.js once

### Description (*)

Continue work that was done by @Nagamaiah007 in #26097.
I changed approach to set events with proper scope of toolbar that javascript is initialized with in order not to process events twice. Previously _bind function was adding events for data-role selectors that were not assigned to this.element scope, so if we had two toolbars on the page (toolbar top and bottom), all events were incorrectly assigned with first toolbar initialization. This change fixes scope for these data-role selectors and won't trigger specific functions to be run twice.

### Related Pull Requests

magento/magento2#26097: Update toolbar.js #25934

### Fixed Issues (if relevant)

1. Fixes magento/magento2#25934

### Manual testing scenarios (*)

1. Got to product list, search or advance search page
2. Open developer tools -> Network tab (Chrome browser)
3. Select/click any of the toolbar option like Grid/List, Sort By, or show X per page
4. Network(Chrome Browser) selected/clicked link triggering two time and because of two time trigger backed doing twice process after canceled one also. (for verify put any log in to catalog->product->list controller).

### Questions or comments


### Contribution checklist (*)
